doc/ref_model/chapters/appendix-a.md

```diff
@@ -42,14 +42,16 @@ As Part of the [Transition Plan](../../gov/chapters/chapter09.md#9.2) described
 ## A.5 Hardware-Dependent Coding Policies
 
-As described in the Principles sections of RM Chapter 1, features that require hardware-dependent code in the workload are prohibited in CNTT compliant implementations. This principle is henceforth referred to as the "Abstraction Principle". Note, this is not the case for the Compute node host software (e.g., host OS). Within the Infra, hosts are expected to have software that is customized for the specific hardware equipped. However, the intent is that these software drivers and higher layers will abstract the Capabilities they enable, thereby exposing them with an open API. An example of exposing capabilities in this manner is implemented in the Virtio family of APIs. This requirement is in support of VNF abstraction and portability of VNFs across the Infra landscape.
+As described in the Principles sections of RM Chapter 1, features that require hardware-dependent code in the workload are prohibited in CNTT compliant implementations. This principle is henceforth referred to as the "Abstraction Principle". Note, this prohibition does not apply to the Compute node host software (e.g., host OS). Within the Infra, hosts are expected to have software that is customized for the specific hardware equipped. However, the intent is that these software drivers and higher layers will abstract the Capabilities they enable, thereby exposing them with an open API. An example of exposing capabilities in this manner is implemented in the Virtio family of APIs. This requirement is in support of VNF abstraction and portability of VNFs across the Infra landscape.
 
-CNTT realizes there are implications to this and limitations to the ability to live by the Abstraction Principle. A textbook example of a Capability that transgresses this principle, is SR-IOV. Other, less notable, yet very important examples include GPU and other acceleration hardware, such as FPGA. A less obvious, yet critically important example is the VNF program(s) itself. As workloads execute native microarchitecture opcodes, those instruction sets effectively constitute an ABI (Application Binary Interface). Additionally, these programs may or may not, attempt to execute vendor-specific extensions to standard instruction sets, such as x86 or ARM.
+CNTT realizes there are implications and limitations to the ability to live by the Abstraction Principle. A textbook example of a Capability that transgresses this principle, is SR-IOV over PCI-PT<sup>1)</sup>. Other, less notable, yet very important examples include GPU and other acceleration hardware, such as FPGA. A less obvious, yet critically important example is the VNF program(s) itself. As workloads are coded in native microarchitecture opcodes, the opcodes' instruction sets effectively constitute an ABI (Application Binary Interface). Additionally, the VNF programs may or may not, attempt to execute vendor-specific extensions to base instruction sets, such as de facto x86 or ARM.
```

Comment:
```suggestion
CNTT realizes there are implications and limitations to the ability to live by the Abstraction Principle. A textbook example of a Capability that transgresses this principle, is SR-IOV over PCI-PT<sup>1)</sup>. Other, less notable, yet very important examples include GPU and other acceleration hardware, such as FPGA. A less obvious, yet critically important example is the VNF program(s) itself. As workloads are coded in native microarchitecture opcodes, the microarchitecture's instruction sets effectively constitute an ABI (Application Binary Interface). Additionally, the VNF programs may or may not, attempt to execute vendor-specific extensions to base instruction sets, such as de facto x86 or ARM.
```
opcodes are part of a machine language instruction, and many of these are what compose an instruction set.
it would be more clear to say 'microarchitecture's instruction set'

Reply:
@kedmison Technically, one could argue that it's the _opcodes_ themselves that comprise the instruction set, as opposed to the operands/aggregate machine instruction, but I take your larger point, as that nuance probably won't resonate w/ most readers... :-) Update in progress.

Reply:
@kedmison Updated. Pls approve (and mark resolved). Thx, -Mark